pkg/kubelet/kubelet.go

```diff
@@ -273,11 +275,30 @@ func makePodSourceConfig(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps
 		glog.Infof("Adding manifest url %q with HTTP header %v", kubeCfg.ManifestURL, manifestURLHeader)
 		config.NewSourceURL(kubeCfg.ManifestURL, manifestURLHeader, nodeName, kubeCfg.HTTPCheckFrequency.Duration, cfg.Channel(kubetypes.HTTPSource))
 	}
+
+	// This is a one-shot fill the pipeline on startup if checkpoints exist
+	if kubeCfg.CheckpointPath != "" {
+		glog.Infof("Adding checkpoint path: %v", kubeCfg.CheckpointPath)
+		cmgr = checkpoint.NewCheckpointManager(kubeCfg.CheckpointPath)
+		c := cfg.Channel(kubetypes.ApiserverSource)
+		pods, err := cmgr.LoadPods()
+		if err != nil {
+			return nil, nil, err
+		}
+		if len(pods) > 0 {
+			c <- kubetypes.PodUpdate{Pods: pods, Op: kubetypes.SET, Source: kubetypes.ApiserverSource}
+			for _, pod := range pods {
+				glog.Infof("Starting Checkpoint %v - Name (%v)", pod.GetUID(), pod.GetName())
+			}
+		}
```

Comment:
Another problem is that if the pod gets deleted from the apiserver while kubelet is down, you would NOT receive a DELETE event from `pkg/kubelet/config/apiserver.go` through the regular channel because the cache in the `pkg/kubelet/config` would be cold. That means you will keep the pods restored from the checkpoint forever. 

That's one of the reasons why I suggested also looking into storing the checkpoints there (`pkg/kubelet/config`).

Reply:
Fixed following your example.